packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -988,7 +1009,23 @@ export class ContainerRuntime
 		return this._getAttachState();
 	}
 
-	public idCompressor: (IIdCompressor & IIdCompressorCore) | undefined;
+	private _idCompressor: (IIdCompressor & IIdCompressorCore) | undefined;
+
+	public get idCompressor() {
+		// Expose ID Compressor only if it's On from the start.
+		// If container uses delayed mode, then we can only expose generateDocumentUniqueId() and nothing else.
+		// That's because any other usage will require immidiate loading of ID Compressor in next sessions in order
+		// to reason over such things as session ID space.
+		if (this.idCompressorMode === "on") {
+			assert(this._idCompressor !== undefined, "compressor should have been loaded");
+			return this._idCompressor;
+		}
+	}
+	protected delayedCompressorLoading = false;
+
+	public generateDocumentUniqueId() {
+		return this._idCompressor?.generateDocumentUniqueId() ?? uuid();
```

Comment:
high entropy uuid generation before load :(
it might be worth a future item to extract just the sequential uuid logic (small amount of code) from the compressor and use it here if the compressor isn't on

Reply:
Maybe? let's make a first step and look at some files / collect some stats. You might be right but notice that at least for DDSs & DataStores I'll still use compact IDs in the case of detached objects (without leveraging this function).
Long term you might be right, but there is also a case to be made to pay the cost for full ID compressor from the start :)